test/new-e2e/tests/agent-platform/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,144 @@
+		"arm64":  e2eOs.ARM64Arch,
+	}
+
+	platformJSON := map[string]map[string]map[string]string{}
+
+	err := json.Unmarshal(platforms.Content, &platformJSON)
+	require.NoErrorf(t, err, "failed to umarshall platform file: %v", err)
+
+	osVersions := strings.Split(*osVersion, ",")
+	cwsSupportedOsVersionList := strings.Split(*cwsSupportedOsVersion, ",")
+	fmt.Println("Parsed platform json file: ", platformJSON)
+	for _, osVers := range osVersions {
+		vmOpts := []ec2params.Option{}
+		osVers := osVers
+		cwsSupported := false
+		for _, cwsSupportedOs := range cwsSupportedOsVersionList {
+			if cwsSupportedOs == osVers {
+				cwsSupported = true
+			}
+		}
+
+		var testOsType ec2os.Type
+		for osName, osType := range osMapping {
+			if strings.Contains(osVers, osName) {
+				testOsType = osType
+			}
+		}
+
+		t.Run(fmt.Sprintf("test upgrade on %s %s", osVers, *architecture), func(tt *testing.T) {
+			tt.Parallel()
+			fmt.Printf("Testing %s", osVers)
+
+			vmOpts = append(vmOpts, ec2params.WithImageName(platformJSON[*platform][*architecture][osVers], archMapping[*architecture], testOsType))
+			if instanceType, ok := os.LookupEnv("E2E_OVERRIDE_INSTANCE_TYPE"); ok {
+				vmOpts = append(vmOpts, ec2params.WithInstanceType(instanceType))
+			}
+
+			e2e.Run(tt, &upgradeSuite{srcVersion: *srcAgentVersion, destVersion: *destAgentVersion, cwsSupported: cwsSupported}, e2e.EC2VMStackDef(vmOpts...), params.WithStackName(fmt.Sprintf("upgrade-from-%s-to-%s-test-%s-%v-%v-%s", *srcAgentVersion, *destAgentVersion, *flavorName, os.Getenv("CI_PIPELINE_ID"), osVers, *architecture)))
+		})
+	}
+}
+
+func (is *upgradeSuite) TestUpgrade() {
+	fileManager := filemanager.NewUnixFileManager(is.Env().VM)
+
+	vm := is.Env().VM.(*client.PulumiStackVM)
+	agentClient, err := client.NewAgentClient(is.T(), vm, vm.GetOS(), false)
+	require.NoError(is.T(), err)
+
+	unixHelper := helpers.NewUnixHelper()
+	VMclient := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
+	is.SetupAgentStartVersion(VMclient)
+	is.UpgradeAgentVersion(VMclient)
+	is.CheckUpgradeAgentInstallation(VMclient)
+
+}
+
+func (is *upgradeSuite) SetupAgentStartVersion(VMclient *common.TestClient) {
+	install.Unix(is.T(), VMclient, installparams.WithArch(*architecture), installparams.WithFlavor(*flavorName), installparams.WithMajorVersion(is.srcVersion), installparams.WithCustomAPIKey(os.Getenv("DATADOG_AGENT_API_KEY")), installparams.WithCustomPipelineID(""))
+	var err error
+	if is.srcVersion == "5" {
+		_, err = VMclient.VMClient.ExecuteWithError("sudo /etc/init.d/datadog-agent stop")
+	} else {
+		_, err = VMclient.SvcManager.Stop("datadog-agent")
+	}
+	require.NoError(is.T(), err)
+}
+
+func (is *upgradeSuite) UpgradeAgentVersion(VMclient *common.TestClient) {
+	var arch string
+	if *architecture == "arm64" && (*platform == "centos" || *platform == "amazonlinux" || *platform == "fedora" || *platform == "redhat") {
+		arch = "aarch64"
+	} else {
+		arch = *architecture
+	}
```

Comment:
Why doing this? The arch param is used to determine the repository to use for downloading the agent package on `deb`. It has no effect on `rpm` installation

Reply:
Ah didn't see it was only used on deb 👍